pkg/apis/workloads/v1alpha1/console_types.go

```diff
@@ -120,6 +120,37 @@ type ConsoleTemplateSpec struct {
 	// +kubebuilder:validation:Minimum=0
 	// +kubebuilder:validation:Maximum=604800
 	DefaultTTLSecondsAfterFinished *int32 `json:"defaultTtlSecondsAfterFinished,omitempty"`
+
+	// List of authorisation rules to match against in order from top to bottom.
+	// +optional
+	AuthorisationRules []ConsoleAuthorisationRule `json:"authorisationRules,omitempty"`
+	// Default authorisation rule to use if no authorisationRules are defined or no authorisationRules match.
+	// +optional
+	DefaultAuthorisationRule *ConsoleDefaultAuthorisationRule `json:"defaultAuthorisationRule,omitempty"`
+}
+
+// ConsoleAuthorisationRule declares rules specifying what commands need to be authorised and by whom.
+type ConsoleAuthorisationRule struct {
+	Name string `json:"name"`
```

Comment:
Missing a comment on this one? It's probably important given its ambiguity otherwise

Reply:
Ah yep, will add this is is shown in audit logs in a comment to direct people on its use.